lib/results_validators/persons_validator.rb

```diff
@@ -74,27 +104,33 @@ def run_validation(validator_data)
         end
         duplicate_newcomer_names = []
         without_wca_id.each do |p|
-          if p.dob.month == 1 && p.dob.day == 1
-            @warnings << ValidationWarning.new(:persons, competition.id,
-                                               DOB_0101_WARNING,
-                                               name: p.name)
-          end
           if p.gender.blank?
             @warnings << ValidationWarning.new(:persons, competition.id,
                                                EMPTY_GENDER_WARNING,
                                                name: p.name)
           end
-          # Competitor less than 3 years old are extremely rare, so we'd better check these birthdate are correct.
-          if p.dob.year >= Time.now.year - 3
-            @warnings << ValidationWarning.new(:persons, competition.id,
-                                               VERY_YOUNG_PERSON_WARNING,
-                                               name: p.name)
-          end
-          if p.dob.year <= Time.now.year - 100
-            @warnings << ValidationWarning.new(:persons, competition.id,
-                                               NOT_SO_YOUNG_PERSON_WARNING,
-                                               name: p.name)
+
+          PersonsValidator.dob_validations(p.dob).each do |validation|
+            case validation
+            when DOB_VALIDATION_ISSUES_MAP[:dob_jan_one]
+              @warnings << ValidationWarning.new(:persons, competition.id,
+                                                 DOB_0101_WARNING,
+                                                 name: p.name)
+            when DOB_VALIDATION_ISSUES_MAP[:dob_very_young]
+              @warnings << ValidationWarning.new(:persons, competition.id,
+                                                 VERY_YOUNG_PERSON_WARNING,
+                                                 name: p.name)
+            when DOB_VALIDATION_ISSUES_MAP[:dob_not_so_young]
+              @warnings << ValidationWarning.new(:persons, competition.id,
+                                                 NOT_SO_YOUNG_PERSON_WARNING,
+                                                 name: p.name)
+            else
+              @warnings << ValidationWarning.new(:persons, competition.id,
+                                                 UNKNOWN_DOB_WARNING,
+                                                 name: p.name)
+            end
```

Comment:
Hm, I'm not a big fan of this "outsourcing" of certain validations.

In general, if some validations are so important that they are being treated differently from the rest (in this case, `dob` validations are being treated very differently / "more special" from the other `Persons` validators) they should ideally have their own validator.
But I understand that this is a deep-reaching change especially regarding the way that WRT works, so I won't force you to go down that path.

But what I do want to avoid is your current mapping. My suggestion: Find a backwards-compatible way to allow passing a more abstract "type" flag into `ValidationWarning` and `ValidationError`. (where "backwards-compatible way" in this case means "way that avoids having to touch _all_ instances of ValidationWarning/ValidationError across _all_ validators in one PR, and just add it optionally to those warnings/errors you care about").
You can then (a) check for the presence (not-null) of the new `type` property, and (b) use that to communicate the error contents to the frontend.

Basically, don't change the core logic of the validators at all, "only" add the additional parameter to `ValidationError` / `ValidationWarning` that allows you to identify the type of warning/error.

Reply:
Okay, I've done as you suggested. But adding a new field to `ValidationIssue` is technically not possible as far as I understood, because the initialization is as follows:

```
def initialize(kind, competition_id, message, **message_args)
```

I definitely need one more field `type` to understand what the error is in the client side. For now I'm living without the `type` field, but it is limiting the features a lot for me. I'll discuss with you over slack and check what can be done for `type` and implement in another PR.